.github/workflows/run_mmperf.yml

```
@@ -0,0 +1,106 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Workflow for running `mmperf` (https://github.com/mmperf/mmperf).
+# `mmperf` benchmarks matrix-multiply workloads on IREE and other backends such
+# as RUY, TVM, Halide, CuBLAS, etc.
+#
+# The workflow runs benchmarks on CPU and GPU and uploads results to the
+# `mmperf-benchmark-artifacts` GC bucket.
+
+name: mmperf
+
+on:
+  workflow_dispatch:
+  pull_request:
+
+jobs:
+  setup:
+    runs-on: ubuntu-20.04
+    env:
+      # The commit being checked out is the merge commit for the PR. Its first
+      # parent will be the tip of main.
+      BASE_REF: HEAD^
+      PR_TITLE: ${{ github.event.pull_request.title }}
+      PR_BODY: ${{ github.event.pull_request.body }}
+    outputs:
+      should-run: ${{ steps.configure.outputs.should-run }}
+      ci-stage: ${{ steps.configure.outputs.ci-stage }}
+      runner-env: ${{ steps.configure.outputs.runner-env }}
+      runner-group: ${{ steps.configure.outputs.runner-group }}
+      write-caches: ${{ steps.configure.outputs.write-caches }}
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
+        with:
+          # We need the parent commit to do a diff
+          fetch-depth: 2
+      - name: "Configuring CI options"
+        id: configure
+        run: |
+          # Just informative logging. There should only be two commits in the
+          # history here, but limiting the depth helps when copying from a local
+          # repo instead of using checkout, e.g. with
+          # https://github.com/nektos/act where there will be more.
+          git log --oneline --graph --max-count=3
+          ./build_tools/github_actions/configure_ci.py
+
+  benchmark_cpu:
+    needs: setup
+    if: needs.setup.outputs.should-run == 'true'
+    runs-on:
+      - self-hosted  # must come first
+      - runner-group=${{ needs.setup.outputs.runner-group }}
+      - environment=${{ needs.setup.outputs.runner-env }}
+      - cpu
+      - os-family=Linux
+    env:
+      MMPERF_RESULTS_DIR: mmperf-results
+      GCS_UPLOAD_DIR: gs://mmperf-benchmark-artifacts
```

This is really a whole bucket, not a directory. In other workflows, we separate artifacts into directories by workflow run. We should consider doing that here also. It looks from the example run like this creates a directory with a specific timestamp. Should a rerun of an old commit really show up with a new timestamp? It looks like `GCS_UPLOAD_DIR` is the same between these jobs, but then you add a suffix. Would it make sense to instead add the suffix here?

```suggestion
      GCS_UPLOAD_DIR: gs://mmperf-benchmark-artifacts/cpu
```

---

I updated the GCS_UPLOAD_DIR to include the cpu or cuda suffix. I also changed the GCS DIR to include the IREE SHA + timestamp instead of the timestamp generated by mmperf.